### PR TITLE
code review

### PR DESCRIPTION
**Pull request by oferisak**
code review 11/10/2023

### CODE REVIEW

File: src/dragen_enrichment_main.R [hunk 1]
```diff
@@ -137,9 +135,9 @@ if (start_from == "run_experiment_name"){
   if (samples_in_project == "each_sample_in_project"){
     # the function also check all fastq are TRUE. If not failed
     pull_run_biosamples(workgroup,run_experiment_name) 
-    pull_run_statistics(workgroup,run_experiment_name, main_output_dir)
-    # The function filter out FALSE fastq
-    option2_each_sample_is_project(run_experiment_name, ht_ref, target_bed_id, baseline_samples_ids, log_dir)
+    pull_run_statistics(workgroup,run_experiment_name, main_output_dir) ## TODO: lets consider getting run statistics in a different script? it feels out of place
```
Comment (BioInfoData) [inline src/dragen_enrichment_main.R:138]: add different script fot run_statistics
change output foldre for run_statistics

File: README.Rmd [hunk 2]
```diff
@@ -40,26 +40,42 @@ For each analysis prepare analysis_set_up file and save it at:
...
 
 **project_dir:** full path to folder "basespace_automation_202206" - should be done only once.
 
 **path_to_selected_parameters:** path to analysis set up file - should be done in each run.
 
+# Analysis modes:
+
 ## 1. Run starting from run sheet:
 
 The location of the run sheet files should be:
```
Comment (BioInfoData) [inline README.Rmd:81]: copy run_sheet also to run folder

File: README.Rmd [hunk 3]
```diff
@@ -40,26 +40,42 @@ For each analysis prepare analysis_set_up file and save it at:
...
+
+### Parameters to change in the code before running the dragen_enrichment_main.R script:
 
 **project_dir:** full path to folder "basespace_automation_202206" - should be done only once.
 
 **path_to_selected_parameters:** path to analysis set up file - should be done in each run.
 
+# Analysis modes:
+
 ## 1. Run starting from run sheet:
```
Comment (BioInfoData) [inline README.Rmd:79]:
```suggestion
## 1. Run starting from run sheet: #test suggestion
```

File: src/dragen_enrichment_main.R [hunk 4]
```diff
@@ -84,11 +82,11 @@ sample_sheet =  select_parameters["sample_sheet","param"] # file must be in fold
 
 input_files = read.csv("./config/general_inputs/input_files_table.csv")
 params = run_dragen_enrichment_preprocess( app_version,genome,kit,cnv, input_files, app_name)
-ht_ref = params[[1]]
+ht_ref = params[[1]] ## TODO
 target_bed_id = params[[2]]
 baseline_samples_ids = params[[3]]
 cnv_baseline_samples_file = params[[4]]
-target_name =  params[[5]]
+target_name =  params[[5]] ## TODO
```
Comment (BioInfoData) [inline src/dragen_enrichment_main.R:90]: assing parameters by name, not number

File: src/dragen_enrichment_main.R [hunk 5]
```diff
@@ -137,9 +135,9 @@ if (start_from == "run_experiment_name"){
   if (samples_in_project == "each_sample_in_project"){
     # the function also check all fastq are TRUE. If not failed
     pull_run_biosamples(workgroup,run_experiment_name) 
-    pull_run_statistics(workgroup,run_experiment_name, main_output_dir)
-    # The function filter out FALSE fastq
-    option2_each_sample_is_project(run_experiment_name, ht_ref, target_bed_id, baseline_samples_ids, log_dir)
+    pull_run_statistics(workgroup,run_experiment_name, main_output_dir) ## TODO: lets consider getting run statistics in a different script? it feels out of place
+    # The function filter out FALSE fastq  ## How do we know that there are missing fastqs? 
+    option2_each_sample_is_project(run_experiment_name, ht_ref, target_bed_id, baseline_samples_ids, log_dir)
```
Comment (BioInfoData) [inline src/dragen_enrichment_main.R:141]: If fastq == F the code will stop

File: README.Rmd [hunk 6]
```diff
@@ -162,7 +178,10 @@ knitr::kable(d, "pipe")
 
 ## Outputs
 
-### Data
+### Data 
+<!---
+this is a bit akward, maybe we want files created by the script to be in the output folder? meaning if were running the script on a sequencing run, then all the biosamples collected will be in the output folder. 
+-->
```
Comment (BioInfoData) [inline README.Rmd:185]: Sample_sheet - to create it in the run output file. To copy to the input_biosample folder and to read from there

File: src/dragen_enrichment_main.R (hunk 5 above)
Comment (BioInfoData) [reply src/dragen_enrichment_main.R:141]: If we continue runnign anyway, only fastq = T will be analyzed

File: README.html
Comment (oferisak) [inline README.html:1]: the html file for the readme shouldnt be in the git. in fact i think there is a way to see the readme you created straight from the repository (so there is no need to create the HTML)

File: src/pull_run_biosamples_main.R
Comment (oferisak) [inline src/pull_run_biosamples_main.R:1]: why is this file called pull run biosamples? it seems like it collects the run statistics (not biosamples)

File: lib/get_dragen_enrichment_stats.R [hunk 7]
```diff
@@ -1,11 +1,11 @@
 
 
-get_dragen_enrichment_stats = function(run_experiment_name, main_output_dir){
+get_dragen_enrichment_stats = function(run_experiment_name, main_output_dir,app_name){
   project_names<-c()
   run_sheet_name_prefix <-run_experiment_name
   project_ids<-get_project_ids_from_runsheet(run_sheet_name_prefix)
   # create a new folder
-  main_output_folder<-glue('{main_output_dir}/basespace_dragen_statistics.{run_sheet_name_prefix}')
+  main_output_folder<-glue('{main_output_dir}/{tolower(gsub(" ", "_", app_name))}.{run_sheet_name_prefix}') ## TODO: lets change this name to [app name in lower case with _ instead of spaces].metrics
```
Comment (oferisak) [inline lib/get_dragen_enrichment_stats.R:8]: please add .metrics at the end of the folder name

File: lib/pull_run_statistics.R [hunk 8]
```diff
@@ -20,11 +20,12 @@ pull_run_statistics = function(workgroup,run_experiment_name, main_output_dir){
   colnames(all_runs_stats)<-stringr::str_replace(colnames(all_runs_stats),'SequencingStats.|SequencingCycles.','')
   
   # save run stats
-  #outfile = glue('./output/run_stats/{workgroup_name}_runstats_runName_{run_experiment_name}.txt')
   outfile = glue('{main_output_dir}/{workgroup_name}_runstats_runName_{run_experiment_name}.txt')
   write.table(all_runs_stats,sep='\t',file=outfile,
               row.names = F,
               quote = F)
+  file.copy(outfile,
+            glue('./output/run_stats/{workgroup_name}_runstats_runName_{run_experiment_name}.txt'))
```
Comment (oferisak) [inline lib/pull_run_statistics.R:28]: not sure we need the "_runName_" 
also as a folder naming convention i like using _ to replace spaces within each section and "." to divide sections. so for example in this folder i would change it to {workgroup_name}.{run_experiment_name}.runstats.txt 
(if everything is separated by _ you dont know where each section starts)

File: lib/run_dragen_enrichment.R [hunk 9]
```diff
@@ -302,9 +310,25 @@ print_params = function(main_output_dir, app_name, app_version, start_from, samp
...
+}
+
+create_project_folders = function(){
+  create_dir("./data/biosamples_sheets")
+  create_dir("./data/biosamples_sheets/runs")
+  create_dir("./data/run_sheets")
+  create_dir("./output")
+  create_dir("./output/run_stats")
+  create_dir("./data/input_biosample_sheets")
+}
```
Comment (oferisak) [inline lib/run_dragen_enrichment.R:334]: not sure i understand, is the create_project_folders function in the run_dragen_enrichment.R script? if so, why? shouldnt they be part of the main initialization script?

File: src/pull_run_biosamples_main.R
Comment (BioInfoData) [reply src/pull_run_biosamples_main.R:1]: mistake in the file name. done

File: lib/get_dragen_enrichment_stats.R (hunk 7 above)
Comment (BioInfoData) [reply lib/get_dragen_enrichment_stats.R:8]: done

File: lib/pull_run_statistics.R (hunk 8 above)
Comment (BioInfoData) [reply lib/pull_run_statistics.R:28]: done

File: README.html
Comment (BioInfoData) [reply README.html:1]: I chnged the html to .md, so now we can see it as the README in github repo

Comment (oferisak) [reply README.html:1]: looks great!! can we remove the README.Rmd now?

**Pull request merged**